File: 백준/Silver/9461. 파도반 수열/파도반 수열.py
import sys

input = sys.stdin.readline


def solve():
    tc = int(input())
    data = []

    for _ in range(tc):
        data.append(int(input()))

    result = [0, 1, 1, 1, 2, 2]  # 1번부터 사용, 0번 인덱스 버림

    max_val = max(data)
    if max_val >= 6:
        result.extend([0] * (max_val - 5))
        for i in range(6, max_val + 1):
            result[i] = result[i - 1] + result[i - 5]

    for num in data:
        print(result[num])


if __name__ == '__main__':
    solve()

File: 백준/Silver/9461. 파도반 수열/파도반 수열.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int t = Integer.parseInt(br.readLine());

		long[] data = new long[101];

		data[1] = 1;
		data[2] = 1;
		data[3] = 1;
		data[4] = 2;
		data[5] = 2;

		for (int i = 6; i <= 100; i++) {
			data[i] = data[i - 1] + data[i - 5];
		}

		StringBuilder sb = new StringBuilder();
		for (int i = 0; i < t; i++) {
			int input = Integer.parseInt(br.readLine());
			sb.append(data[input]).append('\n');
		}

		System.out.print(sb);
	}
}